Migrated k3po.join to k3po.start and k3po.finish

dpwspoon opened the pull request:
The start is required so that robot test code can be written in a single thread when the robot is acting as the client.

jfallows commented on junit/src/main/java/org/kaazing/k3po/junit/rules/SpecificationStatement.java:
```diff
@@ -24,6 +24,7 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
+import java.lang.management.ThreadInfo;
```
Unused?

jfallows commented on junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java:
```diff
@@ -131,5 +132,16 @@ void notifyException(Exception exception) {
+        }
     }
+
+    public void setTestThread(Thread testThread) {
```
Perhaps `setInterruptOnException(Thread interruptible)` ?

jfallows commented on junit/src/main/java/org/kaazing/k3po/junit/rules/SpecificationStatement.java:
```diff
@@ -50,6 +51,8 @@
     public void evaluate() throws Throwable {
 
+        latch.setTestThread(Thread.currentThread());
+
```
Does this need to be cleared at the end, or is it okay just to let it fall out of scope?

dpwspoon replied:
I believe it is fine to fall out of scope

jfallows commented on junit/src/main/java/org/kaazing/k3po/junit/rules/K3poRule.java:
```diff
@@ -130,6 +135,18 @@ public void join() throws Exception {
 
+    public void start() {
+        // script should already be prepared before annotated test can execute
+        assertTrue(format("Did you call join() from outside @%s test?", Specification.class.getSimpleName()), latch.isPrepared());
```
"Did you call **start()** outside ..."

jfallows commented on junit/src/main/java/org/kaazing/k3po/junit/rules/K3poRule.java:
```diff
@@ -130,6 +135,18 @@ public void join() throws Exception {
+
+    public void finish() throws Exception {
+        latch.awaitFinished();
+    }
```
As currently implemented, `k3po.finish()` cannot be used as a direct replacement for `k3po.join()`.  Was that deliberate, or would it be more convenient to let developers be able to call just `k3po.finish()` and let the `k3po.start()` be implicit for that case?

dpwspoon replied:
agreed that it would be

dpwspoon commented:
Thanks for the feed back @jfallows.  

I incorporated your suggestions.  I also updated the parent pom to latest community.  Additionally, I just decided to remove the deprecated join method as this is still a prerelease.
